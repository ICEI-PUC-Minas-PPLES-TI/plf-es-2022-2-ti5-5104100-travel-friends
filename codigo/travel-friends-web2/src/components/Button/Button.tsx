import React from "react";

interface Props {
  bg: string;
  type: "button" | "submit" | "reset" | undefined;
  color: string;
  children?: React.ReactNode;
  onClick?: () => void;
}

const Button: React.FC<Props> = ({ 
    bg,
    type,
    color,
    children,
    onClick,
  }) => { 
  return (
    <button
        type={type}
        onClick={onClick}
        style={{
            backgroundColor: bg,
            border: 'none',
            borderRadius: '6px',
            width: "100%",
            padding: '7px 16px',
            color
        }}
    >
    {children}
    </button>
  );
}

export default Button;
