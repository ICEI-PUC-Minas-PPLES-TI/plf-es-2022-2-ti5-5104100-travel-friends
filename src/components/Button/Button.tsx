import React from "react";

interface Props {
  bg: string;
  type: "button" | "submit" | "reset" | undefined;
  color: string;
  children?: React.ReactNode;
  onClick?: () => void;
  disabled?: boolean;
}

const Button: React.FC<Props> = ({ 
    bg,
    type,
    color,
    children,
    onClick,
    disabled
  }) => { 
  return (
    <button
        type={type}
        onClick={onClick}
        style={{
            backgroundColor: bg,
            border: 'none',
            borderRadius: '6px',
            width: "100%",
            padding: '7px 16px',
            color
        }}
        disabled={disabled}
    >
    {children}
    </button>
  );
}

export default Button;
