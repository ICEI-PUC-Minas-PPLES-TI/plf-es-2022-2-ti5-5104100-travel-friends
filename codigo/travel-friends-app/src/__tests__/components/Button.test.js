import React from "react";
import renderer from "react-test-renderer";

import Button from "../../components/Button";

describe("<Button />", () => {
  it("has 1 child", () => {
    const tree = renderer.create(<Button title="test" />).toJSON();
    expect(tree.children.length).toBe(1);
  });
  it("renders correctly", () => {
    const tree = renderer.create(<Button title="test" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
