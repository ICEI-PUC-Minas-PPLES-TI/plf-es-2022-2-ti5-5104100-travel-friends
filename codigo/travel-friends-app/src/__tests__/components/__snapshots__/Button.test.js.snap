// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> renders correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "alignSelf": "center",
      "backgroundColor": "#3f48cc",
      "borderRadius": 8,
      "justifyContent": "center",
      "opacity": 1,
      "paddingVertical": 10,
      "width": 660,
    }
  }
>
  <Text
    style={
      Object {
        "color": "white",
        "fontSize": 22,
      }
    }
  >
    test
  </Text>
</View>
`;
